pkg/ingestor/collector/collector.go

```diff
@@ -15,4 +15,50 @@
 
 package collector
 
-type Collector interface{}
+import (
+	"context"
+
+	"github.com/artifact-ff/artifact-ff/pkg/config"
+	"github.com/artifact-ff/artifact-ff/pkg/ingestor/collector/gcs"
+	"github.com/artifact-ff/artifact-ff/pkg/ingestor/collector/oci"
+	"github.com/artifact-ff/artifact-ff/pkg/ingestor/collector/pubsub"
+	"github.com/artifact-ff/artifact-ff/pkg/ingestor/collector/transparency"
+	"go.uber.org/zap"
+)
+
+type Collector interface {
+	RetrieveArtifacts(ctx context.Context) (map[string][]byte, error)
```

Comment:
assuming once other PR gets merged, this will change to `map[string]Document` yea?

Reply:
added.